plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsVariationSelector.php

```diff
@@ -308,7 +45,20 @@ protected function render( $attributes, $content, $block ): string {
 		global $product;
 
 		if ( $product instanceof \WC_Product && $product->is_type( 'variable' ) ) {
-			return $this->render_variation_form( $product, $attributes );
+			$variation_attributes = $product->get_variation_attributes();
+
+			if ( empty( $variation_attributes ) ) {
+				return '';
+			}
+
+			$variations = $this->get_variations_data( $product );
+			if ( empty( $variations ) ) {
+				return '';
+			}
+
+			add_filter( 'woocommerce_product_supports', array( $this, 'check_product_supports' ), 10, 3 );
```

Comment:
Not a strong opinion, but if this filter is added to ensure the Product Button knows whether the product can be added to the cart without a page refresh, I’d lean towards modifying the Product Button render method instead.

So instead of checking `$product->supports( 'ajax_add_to_cart' )`, we could assume that adding to the cart without a page refresh is always possible when inside the blockified "Add to Cart with Options" block. What do you think?

Reply:
So are we going to not consider this setting for this case?
<img width="656" alt="image" src="https://github.com/user-attachments/assets/e5ee600e-2d22-4603-a6aa-7fe14f14f8ec" />